doc/langdef.md

```diff
@@ -116,44 +126,185 @@ In general it is a bad idea for those defining contexts or extensions to use
+interpretation.
+
 ## Values
 
 Values in CEL represent any of the following:
 
-1.  *Simple Values*, as they come from protocol buffers: booleans, integers,
-    floating-point numbers, strings, byte strings, and messages. Numbers are
-    normalized to the highest precision supported by protobuf, i.e. int32 as
-    int64, uint32 as uint64, and float as double. Enum values are represented as
-    integers. In addition, the well-known types `Duration` and `Timestamp` are
-    not treated as messages, but as primitive values of the given type.
-2.  *Null Value*, a special value to represent an absent value, for example, the
-    default value of messages.
-3.  *Aggregate Values*, as they are implied by `repeated` and `map` in protobuf
-    messages: lists of the above values, and maps of the above values, with the
-    restriction on allowed map keys implied by protobuf.
-4.  *Type Values*. See [Types](#types) below.
-5.  *Function Values*. See [Functions](#functions) below.
+Type          | Description
+------------- | ---------------------------------------------------------------
+`int`         | 64-bit signed integers
+`uint`        | 64-bit unsigned integers
+`double`      | 64-bit IEEE floating-point numbers
+`bool`        | Booleans (`true` or `false`)
+`string`      | Strings of UTF-8 code points
+`bytes`       | Byte sequences
+`list`        | Lists of values
+`map`         | Associative arrays with `int`, `uint`, `bool`, or `string` keys
+`null_type`   | The value `null`
+message names | Protocol buffer messages
+`type`        | Values representing the types in the first column
+
+### Numeric Values
+
+CEL supports only 64-bit integers and 64-bit IEEE double-precision
+floating-point. We only support positive, decimal integer literals; negative
+integers are produced by the unary negation operator. Note that the integer 7 as
+an `int` is a different value than 7 as a `uint`, which would be written `7u`.
+Double-precision floating-point is also supported, and the integer 7 would be
+written `7.`, `7.0`, `7e0`, or any equivalent representation using a decimal
+point or exponent.
 
-### Dynamic Values
+Note that currently there are no automatic arithmetic conversions for the
+numeric types (`int`, `uint`, and `double`). The arithmetic operators typically
+contain overloads for arguments of the same numeric type, but not for mixed-type
+arguments. Therefore an expression like `1 + 1u` is going to fail to dispatch.
+To perform mixed-type arithmetic, use explicit conversion functions such as
+`uint(1) + 1u`. Such explicit conversions will maintain their meaning even if
+arithmetic conversions are added in the future.
+
+CEL provides no way to control the finer points of floating-point arithmetic,
+such as expression evaluation, rounding mode, or exception handling. However,
+any two not-a-number values will compare equal even if their underlying
+properties are different.
```

Comment:
I think our choice here is slightly different. To roll back on the objective here, we want that representation equality for numbers implies equivalence. For instance if we have list(map(...,list(message-with-double))), we do not want to need to pick apart the entire deep aggregate to determine that some doubles with the same bits are actually not equal because they happen to be NaN.

So I would put it along the representation. "Any two numbers are equal if their representation is equal, even if (in the case of NaN for example) their logical properties may not."

Reply:
In the language council meeting, we decided that all NaN values are unequal, even to themselves, and yes, this implies that even large aggregates are unequal to themselves if they contain a NaN anywhere.  Updating "equal" to "unequal" per the council meeting.